kubespawner/spawner.py

```diff
@@ -583,15 +583,30 @@ def _hub_connect_port_default(self):
     )
 
     def _expand_user_properties(self, template):
-        # Make sure username matches the restrictions for DNS labels
+        # Make sure username and servername match the restrictions for DNS labels
         safe_chars = set(string.ascii_lowercase + string.digits)
-        legacy_escaped_username = ''.join([s if s in safe_chars else '-' for s in self.user.name.lower()])
-        safe_username = escapism.escape(self.user.name, safe=safe_chars, escape_char='-').lower()
```

Comment:
I think we can just add:

```python
name = getattr(self, name, '')
if name:
    servername = '-' + name
else:
    servername = ''

format(
  ...
  servername=servername,
)
```

instead of branching at the higher level.